docs/content/basic/custom-steps.md

```diff
@@ -0,0 +1,103 @@
+---
+title: Listening and responding to custom steps
+lang: en
+slug: /concepts/custom-steps
+---
+
+Your app can use the `function()` method to listen to incoming [custom step requests](https://api.slack.com/automation/functions/custom-bolt). Custom steps are used in Workflow Builder to build workflows. The method requires a step `callback_id` of type string. This `callback_id` must also be defined in your [Function](https://api.slack.com/concepts/manifests#functions) definition. Custom steps must be finalized using the `complete()` or `fail()` listener arguments to notify Slack that your app has processed the request.
+
+* `complete()` requires one argument: an `outputs` object. It completes your custom step **successfully** and provides an object containing the outputs of your custom step as per its definition.
+* `fail()` requires **one** argument: `error` of type string. It completes your custom step **unsuccessfully** and provides a message containing information regarding why your custom step failed.
+
+You can reference your custom step's inputs using the `inputs` listener argument.
+
+```js
+// This sample custom step formats an input and outputs it
+app.function('sample_custom_step', async ({ ack, inputs, complete, fail, logger }) => {
+  try {
+    await ack();
+    const { message } = inputs;
+
+    await complete({
+        outputs: { 
+            message: `:wave: You submitted the following message: \n\n>${message}` 
+        }
+    });
+  } catch (error) {
+    logger.error(error);
+    await fail({ error: `Failed to handle a function request: ${error}` });
+  }
+});
+```
+
+<details>
+  <summary>
+  Listening to custom step actions
+  </summary>
+  Your app can listen to user actions, like button clicks, created from `custom steps` using the `action` method.
```

Comment:
Suggested tweak to the introduction here:
```suggestion
  Your app's custom steps may create interactivity points for users, for example:

- Post a message with a button
- [Open a modal view](creating-modals)

If such interaction points originate from a custom step execution, the events sent to your app representing the end-user interaction with these points are considered to be _function-scoped interactivity events_. These interactivity events can be handled by your app using the same concepts we covered earlier, such as [Listening to actions](action-listening) or [Listening to views](view-submissions).

The one addition that function-scoped interactivity events have is they will contain data related to the `function_executed` event they were spawned from, such as custom step input data and access to `complete()` and `fail()` handlers to manage step control flow.
```

Reply:
This is good 🙇  provides more information without duplicating documentation from api.slack.com

I had to adapt this slightly to because open modal views are not yet supported, but hopefully soon